flexget/components/hook/hook.py

```diff
@@ -0,0 +1,336 @@
+from loguru import logger
+
+from flexget import plugin
+from flexget.task import Task
+from flexget.event import event
+from flexget.config_schema import one_or_more
+from flexget.components.hook.hook_util import ENDPOINT, task_to_dict
+
+
+PLUGIN_NAME = 'hook'
+logger = logger.bind(name=PLUGIN_NAME)
+
+
+class Hook:
+    """
+    Sends a Hook
+
+    Config:
+    hook:
+      title: <<title | optional>>
+      events <<events to lisen [task|phase|plugin] optional>>
+      stages <<stages to lisen [start|end] optional>>
+      tasks  <<tasks to lisen optional>>
+      plugins <<plugins to lisen optional>>
+      when <<when to trigger [accepted|rejected|failed|no_entries|aborted] optional
+      phases <<phases to lisen>>
+      via
+        - <<hook interfaxe>>
+
+    Example:
+      hook:
+        - events:
+            - phase
+            - plugin
+          when: accepted
+          via:
+            - webhooks:
+                host: 'https://noderedurl.1880'
+                method: 'get'
+                endpoint: 'flexget/{{event_tree|join('/')}}'
+                data:
+                  task: '{{task_name}}'
+
+    """
+
+    config = {}
+
+    schema = one_or_more(
+        {
+            'type': 'object',
+            'additionalProperties': False,
+            'properties': {
+                'title': {'type': 'string'},
+                'events': one_or_more(
```

Comment:
This is very, very confusing. These are not our events. Not even by close.

Reply:
The name of the events in this case are simplified, task, are the task events task start and completed! The same for plugin and phases! The idea is to decide if you need to Hook to the specific event and then both stages or just one, for example:

- I just need plugins event's, so I fill event plugin, 
- I just need the end of the plugin's, I fill event plugins and stage end.
- but assume that you just need the plugin movie_exists, you only fill the plugin movie_exists.
- Same logic, if you just need the end of the plugin movie_exists you fill the plugin as movie_exists and the stage as end

This all can be done to phases, or a specific phase, and to task's, or a specific task!

I tried to give as much options as possible, to anyone use as intended!

As stated, all fields are optional, and you can fill as much as you want or need!

If I had used the exact event name, I think it would be more complex and less intuitive, where you would have to know the exact names of the events, that I only know because I looked at the code! Like this I think is more intuitive to know that If I only need to Hook phases, I type phase, and if I only needed to Hook the ending, I type end!

Do you buy this! 😆